### PR TITLE
Query: Add TPT variation for ManyToMany tests

### PR DESCRIPTION
**Pull request by smitpatel**
This also cover some split queries

Part of #21510

### CODE REVIEW

File: test/EFCore.Sqlite.FunctionalTests/Query/TPTManyToManyQuerySqliteTest.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        {
+        }
+
+        // Sqlite does not support Apply operations
+
+        public override Task Skip_navigation_order_by_single_or_default(bool async) => Task.CompletedTask;
+
+        public override Task Filtered_include_skip_navigation_order_by_skip_take_then_include_skip_navigation_where(bool async) => Task.CompletedTask;
+
+        [ConditionalTheory(Skip = "SQLite bug")]
```
Comment (AndriySvyryd) [inline test/EFCore.Sqlite.FunctionalTests/Query/TPTManyToManyQuerySqliteTest.cs:22]: Can this be more specific?

Comment (smitpatel) [reply test/EFCore.Sqlite.FunctionalTests/Query/TPTManyToManyQuerySqliteTest.cs:22]: I copied over from non-TPT M2M thing. I believe underlying sqlite3 had some issue, let me see if I can track it down.

Comment (smitpatel) [reply test/EFCore.Sqlite.FunctionalTests/Query/TPTManyToManyQuerySqliteTest.cs:22]: Filed https://github.com/dotnet/efcore/issues/21541

**Comment by smitpatel** [comment]
@msftbot delay merging this pull request for 2 hours 50 mins

**Pull request merged**